packages/app/pages/index.tsx

```diff
@@ -1,59 +1,62 @@
+    if (!modelCategories) {
+        //? after upgrading to NextJS v15, hitting the back button to return to the homepage results in `undefined` props
+        //? this is a quick fix to reload the page if we encounter that scenario, as modelCategories should never be undefined
+        // TODO: figure out why NextJS v15 is not populating props correctly on back button click
+        window?.location.reload()
+        return null // return null to render nothing until the page refreshes
+    }
+
     return (
         <div>
             <PageTitle title="" />
 
             {(!modelCategories || modelCategories.length < 0) && <UnderMaintenance />}
 
-            {modelCategories?.map(category => (
-                <div key={category.name} className={styles.category}>
-                    <h3>{category.name}</h3>
-
-                    <div className={styles.models}>
-                        {category.models
-                            .sort((a, b) => a.name.localeCompare(b.name))
-                            .map(model => (
-                                <div key={model.name} className={styles.model}>
-                                    <Link href={`/${model.name}`}>
-                                        <Button
-                                            variant="light"
-                                            className="dashboard-model"
-                                        >
-                                            <ModelIcon
-                                                name={model?.icon?.name}
-                                                color={model?.icon?.color}
-                                            />
-                                            <span>{model?.name}</span>
-                                            <span>
-                                                ({model?.['x-meditor']?.count})
-                                            </span>
-                                        </Button>
-                                    </Link>
-                                </div>
-                            ))}
-                    </div>
-                </div>
-            ))}
+            <ModelsByCategory modelCategories={modelCategories} />
         </div>
     )
 }
 
-export async function getServerSideProps(context: NextPageContext) {
+export function sortModelsIntoCategories(models: Model[]): ModelCategory[] {
+    // get a unique list of category names from the models
+    const categories: string[] = models
+        .map(model => model.category) // retrieve just the category name
+        .filter(
+            (category, index, categories) => categories.indexOf(category) === index
+        ) // remove duplicates
+
+    return categories.map(category => ({
+        name: category,
+        models: models.filter(model => model.category === category),
+    }))
+}
+
+export async function getServerSideProps(ctx: NextPageContext) {
```

Comment:
If we ever do move to RSC and away from `getServerSideProps`, there's no more need to serialize data, so we can ditch the JSON parse/stringify hack we currently use: https://github.com/vercel/next.js/issues/11993#issuecomment-1504415523

Reply:
Good to know, I'm not sure we'll get the ability to do it, but it would be nice to remove these hacks